test/functional/legacy/scroll_opt_spec.lua

```diff
@@ -0,0 +1,134 @@
+                                              |
+    ]]
+    local s1 = [[
+      word word word word word word word word |
+      word word word word word word word word |
+      word word word word                     |
+      line three                              |
+      long word long word long word long word |
+      long word long word long word           |
+      ^line                                    |
+      line                                    |
+      line                                    |
+      ~                                       |
+      ~                                       |
+                                              |
+    ]]
+    local s2 = [[
+      word word word word word word word word |
+      word word word word                     |
+      line three                              |
+      long word long word long word long word |
+      long word long word long word           |
+      ^line                                    |
+      line                                    |
+      line                                    |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+                                              |
+    ]]
+    local s3 = [[
+      word word word word                     |
+      line three                              |
+      long word long word long word long word |
+      long word long word long word           |
+      ^line                                    |
+      line                                    |
+      line                                    |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+                                              |
+    ]]
+    local s4 = [[
+      line three                              |
+      long word long word long word long word |
+      long word long word long word           |
+      ^line                                    |
+      line                                    |
+      line                                    |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+                                              |
+    ]]
+    feed('<C-E>')
+    screen:expect(s1)
+    feed('<C-E>')
+    screen:expect(s2)
+    feed('<C-E>')
+    screen:expect(s3)
+    feed('<C-E>')
+    screen:expect(s4)
+    feed('<C-Y>')
+    screen:expect(s3)
+    feed('<C-Y>')
+    screen:expect(s2)
+    feed('<C-Y>')
+    screen:expect(s1)
+    feed('<C-Y>')
+    screen:expect(s0)
+
+    exec('set foldmethod=indent')
+    feed('5G<C-E>')
+    screen:expect(s1)
+    feed('<C-E>')
+    screen:expect(s2)
+    feed('5G<C-Y>')
```

Comment:
Not sure where this cursor discrepancy comes from, the oldtest does `7G`. Not sure it matters either.

Reply:
Ah it comes from `'scrolloff'`. Do we match the vim test by setting scrolloff to 5? Or leave it like this.

Reply:
I think yes.